### ISSUE TITLE
Import and export may appear only with sourcetype module, no import or export statements exist

### ISSUE THREAD

**Issue by mopsyd** [eslint/eslint#11560]
**Tell us about your environment**

* **ESLint Version:** 3.12.0
* **Node Version:**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    “root”: true,
    “env”: {
         “browser”: true,
         “es6”: true
    },
    “parserOptions”: {
        “emcaVersion”: 6
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is a strange one. I added an import statement briefly, which I opted to remove due to sketchy browser support for es6 modules. Ran eslint one time via grunt, got the aforementioned error, which is expected behavior due to my config not having the sourcetype modules entry. This is not the problem.

The problem is that after removing the import line, eslint continues to grumble about that specific file trying to import. There is no import statement. Even with an entirely empty file it still errors. Grepped against the entire project, no import or export statements exist anywhere. At all. My only guess is that it must have cached some faulty reference somewhere, although I have no idea where. May possibly be an upstream issue with gruntify-eslint, although I could not find any specific details or evidence that rules out either that package or this one.

```js
// empty file (previously contained an import line which was removed)
```

**What did you expect to happen?**

Expected that the linter will lint the literal file as is.

**What actually happened? **

Linter sticks on “Import and export may only appear with sourcetype module”. Seems to be holding some memory reference, as no import or export statements exist anywhere in the project.

**Comment by mopsyd** [comment]
Entered this on mobile in transit, will update details when I get home

**Comment by not-an-aardvark** [comment]
Hi, thanks for reporting an issue. If I had to guess, this seems like it's probably an issue with gruntify-eslint not detecting changes to your files. It seems unlikely to me that the problem is coming from ESLint itself, although if you're using the `--cache` flag with ESLint there's a possibility that the issue could be coming from there.

**Comment by mopsyd** [comment]
I had suspected that might be the case. I'll do a bit of investigation when time permits and report either way in case it comes up for someone else again. Nuking the npm node modules folder and pulling it back down again resolved the issue in the interim. I'll have to try to break it again on purpose to diagnose the issue.
